template/9-design.md

```diff
@@ -1,56 +1,193 @@
-- Messaging System: Manages communication between users.
+### User Management
+- **Description**: This functional block handles user-related operations such as registrations, profiles, and interactions between users (e.g., becoming friends).
+- **Application Logic**: Manages user authentication, registration, profile updates, and friend requests.
+- **Framework**: Developed using Firebase Authentication and Firestore for data storage.
+- **Dynamic View Rendering**: Integrates with the frontend to render user profiles and friend lists dynamically based on data from Firestore.
+- **Responsive Backend-Frontend Interaction**: Ensures responsive interaction with the frontend, updating user profiles and friend lists based on user interactions.
+### Event Management
+- **Description**: This block handles operations related to creating, updating, and managing events, including ticket management.
+- **Application Logic**: Manages event creation, updates, and ticket sales.
+- **Framework**: Utilizes Firebase Firestore for event data storage.
+- **Dynamic View Rendering**: Provides APIs to fetch event details dynamically, ensuring frontend displays reflect the latest event information.
+- **Responsive Backend-Frontend Interaction**: Updates event details on the frontend based on user actions such as event creation or ticket purchases.
+### Messaging System
+- **Description**: This block manages communication between users through real-time messaging.
+- **Application Logic**: Handles message sending and retrieval between users.
+- **Framework**: Utilizes Firebase Firestore for real-time messaging.
+- **Dynamic View Rendering**: Updates message threads dynamically based on new messages received.
+- **Responsive Backend-Frontend Interaction**: Ensures real-time updates of message threads on the frontend as new messages are sent or received.
+### Additional Considerations
+- **Monitoring and Logging**: Utilize Firebase Analytics and Firebase Crashlytics to track backend performance and identify issues promptly.
+- **Scalability**: Design the backend to scale horizontally to handle increasing user loads effectively using Firebase's scalability features.
 
 ## Data Model
-
-*What data are you collecting / managing?*
-- User data: Names, email addresses, user's friends, user bios, user profile pictures.
-- Event data: Event locations, categories, start and end date, ticket price, tickets purchased, participant lists, organiser lists.
-
-*How is it organised?*
-- Using firebase database for user and event data.
-- Using firebase storage for QR Code Images, Profile Photos and Event Photos.
-
-*Where is it stored?*
-- Firebase Database
-- Firestore Storage
-
-*How is it shared/copied/cached?*
-- Utilize Firestore’s built-in caching on the client side to reduce latency and enhance the user experience by serving data from cache.
+### Data Collection and Management
+#### User Data
+- **Names:** User's first and last names. (String)
```

Comment:
Nit: Maybe we could change all of these from plural to singular: e.g. Names to name, email addresses to email address, etc to ensure consistency.

Reply:
You're right! Everything was corrected